run/src/run.ts

```diff
@@ -0,0 +1,158 @@
+import { merge, makeReplaySubject } from '@cycle/callbags';
+import { Plugin, Plugins, Main, MasterWrapper, Subscription } from './types';
+import { multicastNow } from './multicastNow';
+
+let currentId = 0;
+
+function cuid(): number {
+  if (currentId === ++currentId) {
+    currentId = 0;
+  }
+  return currentId;
+}
+
+function defaultErrorHandler(err: any): void {
+  throw err;
+}
+
+export function run(
+  main: Main,
+  plugins: Plugins,
+  wrappers: MasterWrapper[],
+  errorHandler: (err: any) => void = defaultErrorHandler
+): Subscription {
+  const masterMain = makeMasterMain(main, plugins, wrappers);
+  checkPlugins(plugins);
+  const connect = setup(plugins, errorHandler);
+  return connect(masterMain);
+}
+
+export function setup(
+  plugins: Plugins,
+  errorHandler: (err: any) => void = defaultErrorHandler
+): (masterMain: Main) => Subscription {
+  checkPlugins(plugins, 'setup', 'First');
+  const { connect, dispose } = setupReusable(plugins, errorHandler);
+
+  return masterMain => {
+    const disconnect = connect(masterMain);
+    return () => {
+      disconnect();
+      dispose();
+    };
+  };
+}
+
+export function setupReusable(
+  plugins: Plugins,
+  errorHandler: (err: any) => void = defaultErrorHandler
+): { connect: (masterMain: Main) => Subscription; dispose: Subscription } {
```

Comment:
Could we give a type `Engine` for the return type of this function? And export it from `./types.ts`

Reply:
done